### PR TITLE
Generic Sensor: Add functionality tests for OrientationSensor

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by tobie** [comment]
Bot ping

**Comment by tobie** [comment]
Bot ping

**Comment by tobie** [comment]
Bot ping

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414815)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414816)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414822)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414823)

# Sauce (MicrosoftEdge) #

**Comment by Honry** [comment]
@alexshalamov, @pozdnyakov, PTAL.

File: orientation-sensor/OrientationSensor.https.html [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    let mat_expect = create_matrix(quat);
+
+    // Works for all supported types.
+    let mat_32 = new Float32Array(16);
+    sensor.populateMatrix(mat_32);
+    assert_array_approx_equals(mat_32, mat_expect, float_precision);
+
+    let mat_64 = new Float64Array(16);
+    sensor.populateMatrix(mat_64);
+    assert_array_approx_equals(mat_64, mat_expect, double_precision);
```
Comment (alexshalamov) [inline orientation-sensor/OrientationSensor.https.html]: why not to use assert_array_equals from testharness? DomMatrix and Float64Array would not lose precision.

File: orientation-sensor/OrientationSensor.https.html [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  sensor.onreading = t.step_func_done(() => {
+    let quat = sensor.quaternion;
+    let mat_expect = create_matrix(quat);
+
+    // Works for all supported types.
+    let mat_32 = new Float32Array(16);
+    sensor.populateMatrix(mat_32);
+    assert_array_approx_equals(mat_32, mat_expect, float_precision);
```
Comment (alexshalamov) [inline orientation-sensor/OrientationSensor.https.html]: Looks like WebAudio has similar helper, maybe it needs to be moved to generic testharness?

File: orientation-sensor/OrientationSensor.https.html (hunk 1 above)
Comment (Honry) [inline orientation-sensor/OrientationSensor.https.html]: We found the test result is sometime failed due to lose precision issue in Float64Array and thought it was same as Float32Array. I will fixed it and file bug if the failure occurs again.

File: orientation-sensor/OrientationSensor.https.html (hunk 2 above)
Comment (Honry) [inline orientation-sensor/OrientationSensor.https.html]: Indeed, I will try to add it to testharness and back to update the test once it is accepted.

Comment (Honry) [inline orientation-sensor/OrientationSensor.https.html]: @alexshalamov, this assertion has been accepted in testharness.js, see https://github.com/w3c/web-platform-tests/pull/6682

**Review by alexshalamov — APPROVED** [review]

**Comment by Honry** [comment]
@alexshalamov, thanks for your review, I will merge it directly.

**Pull request merged**